### PR TITLE
Added QPRK solvers

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [X] Any new documentation only uses public API
  
## Additional context

solves a part of #2177

### CODE REVIEW

**Comment by ParamThakkar123** [comment]
image

@ChrisRackauckas should a separate `alg_order` be defined for QPRK98 ??

File: lib/OrdinaryDiffEqQPRK/src/OrdinaryDiffEqQPRK.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module OrdinaryDiffEqQPRK
+
+import OrdinaryDiffEq: OrdinaryDiffEqAdaptiveAlgorithm, OrdinaryDiffEqConstantCache,
+                       explicit_rk_docstring, @cache, @unpack, OrdinaryDiffEqMutableCache,
+                       OrdinaryDiffEqAdaptiveAlgorithm, @fold, @OnDemandTableauExtract,
+                       trivial_limiter!, alg_cache
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqQPRK/src/OrdinaryDiffEqQPRK.jl:6]:
```suggestion
                       trivial_limiter!, alg_cache, alg_order
```

File: src/OrdinaryDiffEq.jl [hunk 2]
```diff
@@ -288,6 +285,11 @@ include("../lib/OrdinaryDiffEqFIRK/src/OrdinaryDiffEqFIRK.jl")
 using ..OrdinaryDiffEqFIRK
 export RadauIIA3, RadauIIA5, RadauIIA9
 
+include("../lib/OrdinaryDiffEqQPRK/src/OrdinaryDiffEqQPRK.jl")
+using ..OrdinaryDiffEqQPRK
+export QPRK98
+
+include("integrators/integrator_utils.jl")
```
Comment (ChrisRackauckas) [inline src/OrdinaryDiffEq.jl:292]:
```suggestion
```

File: lib/OrdinaryDiffEqQPRK/src/OrdinaryDiffEqQPRK.jl [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module OrdinaryDiffEqQPRK
+
+import OrdinaryDiffEq: OrdinaryDiffEqAdaptiveAlgorithm, OrdinaryDiffEqConstantCache,
+                       explicit_rk_docstring, @cache, @unpack, OrdinaryDiffEqMutableCache,
+                       OrdinaryDiffEqAdaptiveAlgorithm, @fold, @OnDemandTableauExtract,
+                       trivial_limiter!, alg_cache, alg_order
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqQPRK/src/OrdinaryDiffEqQPRK.jl:6]:
```suggestion
                       trivial_limiter!, alg_cache, alg_order, initialize!, perform_step!
```

File: lib/OrdinaryDiffEqQPRK/src/OrdinaryDiffEqQPRK.jl [hunk 4]
```diff
@@ -0,0 +1,18 @@
+module OrdinaryDiffEqQPRK
+
+import OrdinaryDiffEq: OrdinaryDiffEqAdaptiveAlgorithm, OrdinaryDiffEqConstantCache,
+                       explicit_rk_docstring, @cache, @unpack, OrdinaryDiffEqMutableCache,
+                       OrdinaryDiffEqAdaptiveAlgorithm, @fold, @OnDemandTableauExtract,
+                       trivial_limiter!, alg_cache, alg_order, initialize!, perform_step!
+using Static: False
+using MuladdMacro, FastBroadcast
+
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqQPRK/src/OrdinaryDiffEqQPRK.jl:9]:
```suggestion
using RecursiveArrayTools: recursive_unitless_bottom_eltype
```

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas I started a PR for PDIRK and PRK solvers #2303

**Pull request merged**